onnxruntime/core/framework/parallel_executor.cc

```diff
@@ -150,32 +150,36 @@ Status ParallelExecutor::RunNodeAsync(size_t p_node_index,
     // sync before compute
     int queue_id = p_op_kernel->KernelDef().ExecQueueId();
 
-    for (int input_index = 0; input_index < op_kernel_context.InputCount(); ++input_index) {
-      Fence_t fence = op_kernel_context.InputFence(input_index);
-      if (fence) {
-        auto execution_provider_type = p_op_kernel->Node().GetExecutionProviderType();
-        if (OrtMemTypeCPUInput == p_op_kernel->KernelDef().InputMemoryType(input_index)) {
-          execution_provider_type = kCpuExecutionProvider;
+    // Fence is created only if queue_id != 0. Retrieving fence with input_index is costly,
+    // so we can do fence check only when queue_id != 0.
```

Comment:
This comment seems out of date with the code now that queue_id isn't being checked here. Should it be updated?